src/V3Global.cpp

```diff
@@ -39,6 +40,7 @@ void V3Global::boot() {
 
 void V3Global::shutdown() {
     VL_DO_CLEAR(delete m_hierPlanp, m_hierPlanp = nullptr);  // delete nullptr is safe
+    VL_DO_CLEAR(delete m_threadPoolp, m_threadPoolp = nullptr);  // delete nullptr is safe
```

Comment:
What prevents this not to be null at time of shutdown?  Perhaps need an `if (m_threadPoolp) ...`

While there, no harm in doing same to m_hierPlanp.

Reply:
According to C++ spec, delete on nullptr is a noop. Additionally in gcc and clang the `operator delete` calls std::free which also is noop with nullptr. Therefore I believe the null check may be omitted there.